### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by lapenok-aleksej**
1. change structs of database
2. change UI of admin
3. add values of positions
4. admin can add new admin
5. save current year for users

For first start this version I recommend to delete data from database.

### CODE REVIEW

**Comment by mityada** [comment]
1. Why do we need `role_id` in both `user` and `role_users` tables? Isn't it redundant? Same question for `year_id`.
2. Can we find roles by name and not by some magic numbers like [here](https://github.com/NEERC/volunteers/blob/menu/src/main/java/ru/ifmo/neerc/volunteers/controller/AdminController.java#L322-L323)?

**Comment by matvey-kazakov** [comment]
role_users seems to be redundant. It is left by Hibernate from previous runs. I don't see this table in my DB.

**Comment by matvey-kazakov** [comment]
Several items to review:
- Positions > "save" - I don't understand what it does
- Events - how to use edit?
- shall we have bunch of dummy users to play with
- Admin - does not show me while I am admin if I can read that
- Admin > click on name does not do anything
- Need to rearrange UI... Positions are not year specific, halls are year specific...
I think we need to add year name to menu like 2017 events, 2017 halls. and may be split year-specific items and non-year specific items.
- Clicking to year opens strange page with two empty fields
- Year > Back leads nowhere
- Opening event does not show event name at the top.
  I would prefer breadcrumb at the top: Year > Event like 2016 > NEERC

Those items don't stop branch from merge
I can add as separate tickets if you prefer

**Comment by mityada** [comment]
I start server with an empty db and `role_users` is still being created . I guess `mappedBy = "role"` is missed [here](https://github.com/NEERC/volunteers/blob/menu/src/main/java/ru/ifmo/neerc/volunteers/entity/Role.java#L30).

**Comment by mityada** [comment]
BTW trying to add anything except new admin results in exception.

**Comment by matvey-kazakov** [comment]
@mityada you're right, I've not updated when checked. This should be fixed. I mean role_id

**Comment by lapenok-aleksej** [comment]
@mityada `role_users` as I understand need to find all users of current `role` faster, then if we will read all table `user` and find rows where `role=ROLE_USER` or `role=ROLE_ADMIN`

**Comment by mityada** [comment]
@lapenok-aleksej, MySQL (with InnoDB engine) creates indexes for foreign keys automatically, so searching users by `role_id` will not cause a full table scan. No need to worry about performance.

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,5 @@ nbdist/
 
 node_modules
 .sass-cache
-**/webapp/components
```
Comment (artemohanjanyan) [inline .gitignore]: ?

**Comment by lapenok-aleksej** [comment]
Can we merge it?

**Comment by mityada** [comment]
@lapenok-aleksej, I've noticed a strange thing in your commits. When you add some code it often does not contain any spaces [at first](https://github.com/NEERC/volunteers/pull/19/commits/958e12bd8cc8fb5597c60e88f74a009f5891d7b2#diff-51ab96dd27ea6252cd745702766d6610R175), but then, spaces are added in [one of the following commits](https://github.com/NEERC/volunteers/commit/1302e180a0e151b367d9f286d762636fb2c4fe9d#diff-51ab96dd27ea6252cd745702766d6610L175). I wonder, is it the result of automatic code formatting? If so, I think it's better to do it in the same commit where the code was added, otherwise that changed lines make the following diffs hard to read.

**Comment by lapenok-aleksej** [comment]
Yes, it is the result of automatic code formatting. Sometimes I forget to run it before commit.

**Pull request merged**